src/Elasticsearch.Net/Utf8Json/Attributes.cs

```diff
@@ -0,0 +1,58 @@
+#region Utf8Json License https://github.com/neuecc/Utf8Json/blob/master/LICENSE
+// MIT License
+//
+// Copyright (c) 2017 Yoshifumi Kawai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+#endregion
+
+using System;
+
+namespace Elasticsearch.Net
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+	internal class JsonFormatterAttribute : Attribute
+    {
+        public Type FormatterType { get; private set; }
+        public object[] Arguments { get; private set; }
+
+        public JsonFormatterAttribute(Type formatterType)
+        {
+            this.FormatterType = formatterType;
+        }
+
+        public JsonFormatterAttribute(Type formatterType, params object[] arguments)
+        {
+            this.FormatterType = formatterType;
+            this.Arguments = arguments;
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Constructor, AllowMultiple = false, Inherited = true)]
+	internal class SerializationConstructorAttribute : Attribute
+    {
+
+    }
+
+    [AttributeUsage(AttributeTargets.Interface, AllowMultiple = false, Inherited = false)]
```

Comment:
A small summary would be good for this one since its used in our domain poco's quite a bit